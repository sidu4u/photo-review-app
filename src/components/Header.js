import React from "react";
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledDiv = styled.div`
font-size:large;
font-weight:600;
color:blue;
`

const Header = ({data,customClass})=>{
    return (<StyledDiv id="header" className={customClass}>{data}</StyledDiv>);
}

Header.propTypes={
    data:PropTypes.string,
    customClass:PropTypes.string
}

export default Header;