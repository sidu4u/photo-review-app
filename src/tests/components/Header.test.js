import React from "react";
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

const data = "sample data1";
let component;

beforeEach(() => {
    component = shallow(<Header data={data}/>);
  })
  
  afterEach(() => {
    component = null;
  });

it('renders correctly',()=>{
    const component = renderer
    .create(<Header data={data}/>)
    .toJSON();
    expect(component).toMatchSnapshot();
});

it('renders passed pros correctly',()=>{
    expect(component.find("#header").text()).toBe(data);
});