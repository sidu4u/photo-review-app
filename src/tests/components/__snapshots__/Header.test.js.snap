// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdnxRM fWCeML"
  id="header"
>
  sample data1
</div>
`;
